include/uri/UAuthority.h

```diff
@@ -0,0 +1,291 @@
+                           "domain='" + (domain_.empty() ? "null" : domain_) + "', " +
+                           "address=" + (address_.empty() ? "null" : address_) + ", " +
+                           "markedRemote=" + (markedRemote_ ? "true" : "false") + ", " +
+                           "markedResolved=" + (markedResolved_ ? "true" : "false") + "}";
+    }
+
+ private:
+    /**
+     * Constructor for building a UAuthority.
+     * @param device        The device a software entity is deployed on, such as the VCU, CCU or cloud provider.
+     * @param domain        The domain a software entity is deployed on, such as vehicle or Cloud (PaaS).
+     * @param address       The device IP address of the device.
+     * @param markedRemote  Indicates if this UAuthority was implicitly marked as remote.
+     * @param markedResolved Indicates that this uAuthority has all the information needed to be serialised
+     *                          in the long format or the micro format of a UUri.
+     */
+    UAuthority(const std::string& device,
+               const std::string& domain,
+               const std::string& address,
+               const bool markedRemote,
+               const bool markedResolved)
+               : markedRemote_(markedRemote), markedResolved_(markedResolved) {
+        if (!device.empty()) {
+            auto current = isBlank(device) ? "" : device;
+            std::transform(current.begin(), current.end(), current.begin(),
+                           [](unsigned char c) { return std::tolower(c); });
+            this->device_ = std::move(current);
+        }
+        if (!domain.empty()) {
+            auto current = isBlank(domain) ? "" : domain;
+            std::transform(current.begin(), current.end(), current.begin(),
+                           [](unsigned char c) { return std::tolower(c); });
+            this->domain_ = std::move(current);
+        }
+        address_ = isBlank(address) ? "" : address;
+        // To have a unified format for ipv6 addressing.
+        try {
+            if (std::array<uint8_t, sizeof(struct in6_addr)> ipv6{0}; inet_pton(AF_INET6, address.c_str(), &ipv6) == 1) {
+                std::string ipv6str(INET6_ADDRSTRLEN, '\0');
+                if (inet_ntop(AF_INET6, &ipv6, ipv6str.data(), INET6_ADDRSTRLEN) != nullptr) {
+                    address_ = ipv6str.data();
+                } else {
+                    address_ = "";
+                }
+            }
+        } catch (const std::invalid_argument& e) {
+            spdlog::error("Invalid IP: {}, {}", address, e.what());
+        }
+    }
+
+    /**
+     * Utility method to verify if the string is blank.
+     * @param str   The string to be checked
+     * @return bool Returns true if the string is blank.
+     */
+    [[nodiscard]] static bool isBlank(std::string_view str) {
+        return std::all_of(str.begin(), str.end(), isspace);
```

Comment:
This part of code stopped compiling - i think it should be 

 return std::all_of(str.begin(), str.end(), [](char c) { return std::isspace(c) || c == '\0'; });